packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneAttachmentBlobs.spec.ts

```diff
@@ -589,4 +593,244 @@ describeNoCompat("GC attachment blob tombstone tests", (getTestObjectProvider) =
+            // de-duped and redirect to the same storageId.
+            mainContainer.disconnect();
+            const blobContents = "Blob contents";
+            const localHandle1 = await mainDataStore._context.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+            const localHandle2 = await mainDataStore._context.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+            // Connect the container after the blob is uploaded. Send an op to transition the container to write mode.
+            mainContainer.connect();
+            mainDataStore._root.set("transition to write", "true");
+            await ensureContainerConnectedWriteMode(mainContainer);
+
+            // Add the blob's local handles to reference them.
+            mainDataStore._root.set("localBlob1", localHandle1);
+            mainDataStore._root.set("localBlob2", localHandle2);
+
+            // Upload the same blob. This will get de-duped and we will get back a handle with the storageId instead of
+            // the localId that we got when uploading in detached container.
+            const storageHandle = await mainDataStore._context.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+            assert.notStrictEqual(
+                localHandle1.absolutePath, storageHandle.absolutePath, "local and storage handles should be different");
+
+            // Remove the blob's local handles to unreference them.
+            mainDataStore._root.delete("localBlob1");
+            mainDataStore._root.delete("localBlob2");
+
+            // Summarize so that the above attachment blobs are marked unreferenced.
+            await provider.ensureSynchronized();
+            await summarizeNow(summarizer);
+
+            // Wait for sweep timeout so that the blob is tombstoned.
+            await delay(sweepTimeoutMs + 10);
+
+            // Send an op to update the current reference timestamp that GC uses to make sweep ready objects.
+            mainDataStore._root.set("key", "value");
+            await provider.ensureSynchronized();
+
+            // Summarize so that the tombstoned blobs are now part of the summary.
+            const summary2 = await summarizeNow(summarizer);
+
+            // Load a new container from the above summary which should have the blobs tombstoned.
+            const container2 = await loadContainer(summary2.summaryVersion);
+
+            // Retrieving the blob via any of the handles should fail. Note that the blob is requested via its url since
+            // this container does not have access to the blob's handle.
+            const localResponse1 = await container2.request({ url: localHandle1.absolutePath });
+            assert.strictEqual(localResponse1?.status, 404, `Expecting a 404 response for local handle 1`);
+            assert(localResponse1.value.startsWith("Blob removed by gc:"), `Unexpected value for local handle 1`);
```

Comment:
Should we be worried about the stability of this response? Would the response code be enough to assert what we need?

Reply:
Checking the value is basically differentiating between cases where the blob never existed vs where it is deteted by GC. Currently, this is only way to do that. We will be adding something much more deterministic and we will change it then.